rust/core-lib/src/view.rs

```diff
@@ -927,29 +935,46 @@ impl View {
         self.find_changed = FindStatusChange::All;
         self.set_dirty(text);
 
-        // todo: this will be changed once multiple queries are supported
-        // todo: for now only a single search query is supported however in the future
-        // todo: the correct Find instance needs to be updated with the new parameters
-        if self.find.is_empty() {
-            self.find.push(Find::new());
+        // set selection as search query for first find if no additional search queries are used
+        // otherwise add new find with selection as search query
+        if self.find.len() != 1 {
+            self.add_find();
         }
 
-        self.find.first_mut().unwrap().do_find(text, search_query, case_sensitive, false, true);
+        self.find.last_mut().unwrap().do_find(text, search_query, case_sensitive, false, true);
     }
 
-    pub fn do_find(&mut self, text: &Rope, chars: String, case_sensitive: bool, is_regex: bool,
-                   whole_words: bool) {
+    fn add_find(&mut self) {
+        let id = self.find_id_counter.next();
+        self.find.push(Find::new(id));
+    }
+
+    pub fn do_find(&mut self, text: &Rope, queries: Vec<FindQuery>) {
         self.set_dirty(text);
         self.find_changed = FindStatusChange::Matches;
 
-        // todo: this will be changed once multiple queries are supported
-        // todo: for now only a single search query is supported however in the future
-        // todo: the correct Find instance needs to be updated with the new parameters
-        if self.find.is_empty() {
-            self.find.push(Find::new());
-        }
+        // remove deleted queries
+        self.find.retain(|f| queries.iter().position(|q| q.id == Some(f.id())).is_some());
```

Comment:
I did not until now know that this method existed on `Vec<_>`. I think in other places I've done a filter and collect to achieve this.